src/join.ts

```diff
@@ -95,35 +100,125 @@ function parseCallConfig(callConfig: string): DailyCallOptions {
   }
 }
 
+// getModifySdpHook takes a desired codec and returns the given hook to pass
+// to Daily to prefer that codec.
+function getModifySdpHook(wantedCodec: string): ((rtcSDP: any) => any) | null {
+  if (wantedCodec === '') {
+    return null;
+  }
+  const valid = ['VP8', 'VP9', 'H264'];
+  const codecName = wantedCodec.toUpperCase();
+  let found = false;
+  for (let i = 0; i < valid.length; i += 1) {
+    const c = valid[i];
+    if (c === codecName) {
+      found = true;
+      break;
+    }
+  }
+  if (!found) {
+    throw new Error(
+      `invalid codec name supplied: ${wantedCodec}; valid options are: ${valid.join(
+        ' '
+      )}`
+    );
+  }
+  const hook = (rtcSDP: any) => {
+    try {
+      const camIdx = 0;
+      const parsed = sdpTransform.parse(rtcSDP.sdp);
+      const camMedia = parsed.media[camIdx];
+      const preferredCodec = camMedia.rtp.filter(
+        (r: any) => r.codec === codecName
+      );
+      const notPreferredCodec = camMedia.rtp.filter(
+        (r: any) => r.codec !== codecName
+      );
+      const newPayloads = [...preferredCodec, ...notPreferredCodec]
+        .map((r) => r.payload)
+        .join(' ');
+      parsed.media[camIdx].payloads = newPayloads;
+      const newSdp = sdpTransform.write(parsed);
+      return newSdp;
+    } catch (e) {
+      console.error(`failed to set codec preference: ${e}`);
+    }
+    return rtcSDP;
+  };
+  return hook;
+}
+
 // buildCallOptions takes a call config JSON string
 // and returns an associated instance of DailyCallOptions
-function buildCallOptions(callConfig: string): DailyCallOptions {
+function buildCallOptions(
+  callConfig: string,
+  preferredCodec: string = ''
+): DailyCallOptions {
   const callOptions = parseCallConfig(callConfig);
+  const hook = getModifySdpHook(preferredCodec);
 
   const defaultCallOptions = <DailyCallOptions>{
     dailyConfig: {
       avoidEval: true,
+      modifyLocalSdpHook: hook,
     },
   };
+
   const merged = merge(defaultCallOptions, callOptions);
   return merged;
 }
 
 // createCallObject creates a Daily call object with the given
 // configuration and returns it in the form of a DailyCall
-function createCallObject(callConfig: string): DailyCall {
-  const callOptions = buildCallOptions(callConfig);
+function createCallObject(
+  callConfig: string,
+  wantedCodec: string = ''
+): DailyCall {
+  const callOptions = buildCallOptions(callConfig, wantedCodec);
   const co = DailyIframe.createCallObject(callOptions);
   return co;
 }
 
+function getBandwidth(bandwidthConfig: string): Bandwidth {
+  try {
+    const parsedConfig = JSON.parse(bandwidthConfig);
+    const bandwidth = <Bandwidth>parsedConfig;
+
+    // Some very rudimentary validation, since
+    // Daily doesn't seem to throw an error if this
+    // is bad like it does with call object creation
+    const { kbs } = bandwidth;
+    if (kbs && typeof kbs !== 'number' && kbs !== 'NO_CAP') {
```

Comment:
Hmmmmmm should it be `kbs != 'NO_CAP'` ? If you're trying to compare two strings I think `===` will return false if the two strings contents are the same but `==` returns true. https://stackoverflow.com/questions/3586775/what-is-the-correct-way-to-check-for-string-equality-in-javascript

Reply:
I don't _think_ so. There's no reason to allow type coercion here imo, since we just want a strict string comparison.